File: 1716/chipD.py
import math

n,k = [int(x) for x in input().split()]

numways = [0 for i in range(n+1)]
numways[0] = 1
total = [0 for i in range(n+1)]
# for step in range(k,int(math.sqrt(2*n)+5)):
#     print(step)


step = k

while(step*(step+1) -k*(k+1) < 2*n):
    # print(step)
    newnumways = [0 for i in range(n+1)]

    for i in range(n+1):
        adder = i+step
        while(adder<=n):
            newnumways[adder] += numways[i]
            adder += step

        total[i] += newnumways[i]

    numways = newnumways
    step +=1


for i in range(1,n+1):
    print(total[i], end=' ')
print()

File: 1716/chipD.cpp
#include <iostream>
#include <math.h>

int main() {
  long n,k;
  std::cin >> n >> k;

  long numways [n+1];
  long total [n+1];
  for(int i=0; i<n+1; i++){
    numways[i] = 0;
    total[i] = 0;
  }
  numways[0] = 1;


  long step = k;
//464662708
  // while((step-5)*(step+1-5)-k*(k+1)< 2*n) {
  while(step < 10+sqrt(2*n+k*(k+1))) {
    long newnumways [n+1];
    for(int i=0;i<n+1; i++){
      newnumways[i] = 0;
    }

    for(int i=0; i<n+1; i++){
      int adder = i+step;
      while(adder<= n){
        newnumways[adder] = (newnumways[adder]+numways[i])%998244353;
        adder+=step;
      }
      total[i] = (total[i]+newnumways[i])%998244353;

    }
    // numways = newnumways.copy;
    for(int i=0;i<n+1;i++){
      numways[i] = newnumways[i];
    }

    step+=1;
  }
  for(int i=1; i<=n; i++){
    std::cout << (total[i]%998244353) << " ";
  }



}
